modules/experimental-layers/src/tile-layer/utils/tile.js

```diff
@@ -0,0 +1,43 @@
+export default class Tile {
+  constructor({getTileData, x, y, z}) {
+    this.x = x;
+    this.y = y;
+    this.z = z;
+
+    this._data = null;
+    this._loader = null;
+    this.isVisible = false;
+
+    this.getTileData = getTileData;
+
+    this._loader = this._loadData();
+  }
+
+  get data() {
+    if (this._data) {
+      return Promise.resolve(this._data);
+    }
+    return this._loader;
+  }
+
+  get isLoaded() {
+    return this._data !== null;
+  }
+
+  overlaps(tile) {
+    const {x, y, z} = this;
+    const m = Math.pow(2, tile.z - z);
+    return Math.floor(tile.x / m) === x && Math.floor(tile.y / m) === y;
+  }
+
+  _loadData() {
+    const {x, y, z} = this;
+    return (
+      this.getTileData &&
+      this.getTileData({x, y, z}).then(buffers => {
+        this._data = buffers;
+        return buffers;
+      })
```

Comment:
Need to handle loading errors

Reply:
Is logging the stack trace enough? Or do we have other mechanism to handle errors.

Reply:
Consider the case where you don't have tiles above MAX_ZOOM but you can still allow users to zoom in. When the tiles at the current zoom level fail to load (either throw errors or resolve to null), you need to mark the tiles as invalid and fallback to using data from an available zoom level.

Let's land this layer as is and address this in a follow up PR.

Reply:
Yep. Added a TODO comment.